src/apple/macos/ffi.rs

```diff
@@ -154,3 +177,133 @@ mod io_service {
 mod io_service {}
 
 pub use io_service::*;
+
+#[repr(C)]
+pub struct __IOHIDServiceClient(libc::c_void);
+
+pub type IOHIDServiceClientRef = *const __IOHIDServiceClient;
+
+#[repr(C)]
+pub struct __IOHIDEvent(libc::c_void);
+
+pub type IOHIDEventRef = *const __IOHIDEvent;
+
+pub const kIOHIDEventTypeTemperature: i64 = 15;
+
+pub const kIOHIDEventTypePower: i64 = 15;
+
+pub fn IOHIDEventFieldBase(event_type: i64) -> i64 {
+    event_type << 16
+}
+
+#[cfg(not(feature = "apple-sandbox"))]
+extern "C" {
+    pub fn IOHIDEventSystemClientCreate(allocator: CFAllocatorRef) -> *mut libc::c_void;
+    
+    pub fn IOHIDEventSystemClientSetMatching(client: *mut libc::c_void, matches: CFDictionaryRef) -> i32;
+
+    pub fn IOHIDEventSystemClientCopyServices(client: *mut libc::c_void) -> CFArrayRef;
+
+    pub fn IOHIDServiceClientCopyProperty(service: IOHIDServiceClientRef, key: CFStringRef) -> CFStringRef;
+
+    pub fn IOHIDServiceClientCopyEvent(service: IOHIDServiceClientRef, v0: i64, v1: i32, v2: i64) -> IOHIDEventRef;
+
+    pub fn IOHIDEventGetFloatValue(event: IOHIDEventRef, field: i64) -> f64;
+}
+
+pub type CFMutableArrayRef = *mut __CFArray;
+
+extern "C" {
+    pub fn CFArrayCreateMutable(allocator: CFAllocatorRef, capacity: CFIndex, callbacks: *const CFArrayCallBacks) -> CFMutableArrayRef;
+
+    pub fn CFArrayAppendValue(the_array: CFMutableArrayRef, value: *const libc::c_void);
+}
+
+pub(crate) const HID_DEVICE_PROPERTY_VENDOR_ID: &str = "VendorId";
```

Comment:
A lot of these constants don't seem to be used. Please remove the extra ones (or comment them, as you prefer).